lib/utils.pm

```diff
@@ -1685,6 +1686,38 @@ sub create_btrfs_subvolume {
     assert_script_run("rm -fr /tmp/arm64-efi/");
 }
 
+=head2 create_raid_loop_device 
+    
+ create_raid_loop_device([raid_type => $raid_type], [device_num => $device_num], [file_size => $file_size]);
+
+Create a raid array over loop devices.
+Raid type is C<$raid_type>, using C<$device_num> number of loop device, 
+with the size of each device being C<$file_size> megabytes.
+
+Example to create a RAID C<5> array over C<3> loop devices, C<200> Mb each:
+    create_raid_loop_device(raid_type => 5, device_num => 3, file_size => 200)
+
+=cut
+sub create_raid_loop_device {
+    my %args         = @_;
+    my $raid_type    = $args{raid_type}  // 1;
+    my $device_num   = $args{device_num} // 2;
+    my $file_size    = $args{file_size}  // 100;
+    my $loop_devices = "";
+
+    for my $num (1 .. $device_num) {
+        my $raid_file = "raid_file" . $num;
+        assert_script_run("touch $raid_file");
```

Comment:
Don't need to `touch` the file, dd will create it. Btw fallocate is much faster than dd.
I'm just curious and have feeling it's getting unnecessary complicated. Why mdmonitor, can't be simpler service used ?

Reply:
> Don't need to `touch` the file, dd will create it. Btw fallocate is much faster than dd.
Thanks! will do that!

> OT: I'm just curious and have feeling it's getting unnecessary complicated. Why mdmonitor, can't be simpler service used ?
checking mdmonitor is part of the systemd testsuite, I m not keen on this either but it is so that qa_test_systemd does not fail.

Reply:
the thing is at the beginning of the testsuite's life it must have been pretty simple. Just restarting a couple of services to see that they run. Now increasingly those services need setups otherwise they fail

Reply:
I know, we have to do what we have to do. :smile: